server/redux/reducers/user-reducer.js

```diff
@@ -37,14 +37,16 @@ const removeUser = userId => {
 
 /* --------------- THUNK ACTION CREATORS --------------- */
 
-const createAndEmitUser = socket => {
+const createAndEmitUser = (socket, user) => {
   return dispatch => {
     const userId = socket.id;
-    const user = Map(createUser(userId));
```

Comment:
This createUser seems a bit confusing to me. Is there a reason we can't just export and use the User constructor directly to make things a bit clearer?